src/main/kotlin/com/exactpro/th2/rptdataprovider/services/cradle/MessageGroupCradleService.kt

```diff
@@ -150,56 +177,118 @@ class MessageGroupCradleService(
-                K_LOGGER.info { "'$sessionAlias' session alias isn't in [${interval.start}, ${interval.end}] interval" }
-                return@logTime null
+        val shortInterval = Interval(counter.frameStart, counter.frameStart.plusMillis(100))
+        return seachSessionGorupByGroupedMessage(sessionAlias, shortInterval, bookId, cache).also {
+            if (it == null) {
+                K_LOGGER.warn { "Mapping between a session group and the '${sessionAlias}' alias isn't found by statistics, book: ${bookId.name}, interval: [${interval.start}, ${interval.end}] " }
             }
+        }
+    }
 
-            val shortInterval = Interval(counter.frameStart, counter.frameStart.plusMillis(100))
-            K_LOGGER.debug { "Strat searching session group by '$sessionAlias' alias in cradle in (${shortInterval.start}, ${shortInterval.end}) interval" }
-            storage.getSessionGroupsAsync(bookId, shortInterval).await().asSequence()
-                .flatMap { group ->
-                    storage.getGroupedMessageBatches(
-                        GroupedMessageFilter.builder()
-                            .bookId(bookId)
-                            .timestampFrom().isGreaterThanOrEqualTo(shortInterval.start)
-                            .timestampTo().isLessThanOrEqualTo(shortInterval.end)
-                            .groupName(group)
-                            .build()
-                    ).asSequence()
-                }.forEach searchInBatch@{ batch ->
-                    cache.get(sessionAlias)?.let { group ->
-                        K_LOGGER.debug { "Another coroutine has dound '$sessionAlias' session alias to '$group' group pair" }
-                        return@logTime group
-                    }
-                    K_LOGGER.debug { "Search session group by '$sessionAlias' alias in grouped batch" }
-                    batch.messages.forEach { message ->
-                        cache.putIfAbsent(message.sessionAlias, batch.group) ?: run {
-                            K_LOGGER.info { "Put '${message.sessionAlias}' session alias to '${batch.group}' group to cache" }
-                            if (sessionAlias == message.sessionAlias) {
-                                K_LOGGER.debug { "Found '${message.sessionAlias}' session alias to '${batch.group}' group pair" }
-                            }
-                        }
-                    }
-                }
+    private suspend fun seachSessionGorupByPage(
+        sessionAlias: String,
+        interval: Interval,
+        bookId: BookId,
+        cache: Cache<String, String>
+    ): String? {
+        K_LOGGER.debug { "Strat searching '$sessionAlias' session alias in cradle in [${interval.start}, ${interval.end}] interval by page" }
+        // getPagesAsync method is used instead of getPage because the first one return all pages touched by interval
+        val pageInterval = storage.getPagesAsync(bookId, interval).await().asSequence()
```

Comment:
How is it possible that method that supposes to do the same work but asynchronously does work different?

Reply:
I have reverted this change in cradle. I was for test